### ISSUE TITLE
flycheck-mypy unexpected error

### ISSUE THREAD

**Issue by xuhdev** [flycheck/flycheck#1815]
Thank you for taking the time to report an issue and improve Flycheck.  This template is for **actual bugs you observed**.  If you have trouble setting up Flycheck, or if you have a question, please use the relevant issue template instead.

## Checklist
- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description

Although mypy works properly on the command line, `flycheck-mypy` produce errors.

## Steps to reproduce
- Make sure mypy is installed and you are not in a Python virtual environment
- `emacs test.py`, write something and save
- the following error is shown
```
Suspicious state from syntax checker python-mypy: Flycheck checker python-mypy returned 2, but its output contained no errors: unknown option --show-column-numbers
usage: /usr/bin/python3 [option] ... [-c cmd | -m mod | file | -] [arg] ...
Try `python -h' for more information.

Try installing a more recent version of python-mypy, and please open a bug report if the issue persists in the latest release.  Thanks!
```

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## System configuration
```
Syntax checkers for buffer export-account-totals.py in python-mode:

First checker to run:

  python-flake8
    - may enable:         yes
    - executable:         Found at /usr/bin/python3
    - configuration file: Not found
    - `flake8' module:    Found at "/home/hong/.local/lib/python3.7/site-packages/flake8/__init__.py"
    - next checkers:      python-pylint, python-mypy

Checkers that may run as part of the first checker's chain:

  python-mypy
    - may enable:         yes
    - may run:            t
    - executable:         Found at /usr/bin/python3
    - configuration file: Not found

Checkers that could run if selected:

  python-pycompile  select
    - may enable:    yes
    - executable:    Found at /usr/bin/python3
    - next checkers: python-mypy

Checkers that are compatible with this mode, but will not run until properly configured:

  python-pylint (automatically disabled) reset
    - may enable:         no
    - executable:         Found at /usr/bin/python3
    - configuration file: Not found
    - `pylint' module:    Missing; sys.path is ['/usr/lib/python37.zip', '/usr/lib/python3.7', '/usr/lib/python3.7/lib-dynload', '/home/hong/.local/lib/python3.7/site-packages', '/usr/local/lib/python3.7/dist-packages', '/usr/lib/python3/dist-packages']
    - next checkers:      python-mypy

The following syntax checkers are not registered:
  - lsp
Try adding these syntax checkers to `flycheck-checkers'.

Flycheck Mode is enabled. Use SPC u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 32snapshot (package: 20200610.1809)
Emacs version:    27.0.91
System:           x86_64-pc-linux-gnu
Window system:    x
```

Emacs configuration:
- [ ] Plain Emacs / Custom configuration
- [x] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

## Additional notes

If I install mypy in a Python virtual environment, and start Emacs inside that environment, this issue is gone.

[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

**Comment by cpitclaudel** [comment]
Thanks. Did you Try installing a more recent version of python-mypy?

**Comment by xuhdev** [comment]
Yes. I also tried uninstall and reinstall the most recent version of python-mypy. Everything works fine on command line.

**Comment by xuhdev** [comment]
As my suspicion, from the error message, it doesn't seem like `mypy` is called. For some reason seems like `python3` is called, but I'm not sure how to dig out which commands were actually called.

**Comment by cpitclaudel** [comment]
Use `C-c ! C-c` and select mypy; it will show you the command that it's running.  It's a bit weird that you get this python error instead of a mypy one, indeed.

**Comment by xuhdev** [comment]
I explicitly ran `python-mypy` checker and it output the same error.

**Comment by cpitclaudel** [comment]
Perfect; can you show the output?

**Comment by xuhdev** [comment]
```
( 74/123) Run syntax checker as compile command: 

processing
proselint
protobuf-protoc
protobuf-prototool
pug
puppet-lint
puppet-parser
python-flake8
python-mypy
python-pycompile
python-pylint
r-lintr
racket
rpm-rpmlint
```

(`python-flake8` is selected)

**Comment by cpitclaudel** [comment]
Sorry, that's not what I meant.  I was hoping to see the buffer that is shown after you select python-mypy in that list.

**Comment by xuhdev** [comment]
It's exactly the same as the original comment:

```
-*- mode: compilation; default-directory: "~/tmp/" -*-
Compilation started at Fri Jul 31 21:21:40

/usr/bin/python3 --show-column-numbers /home/hong/tmp/test.py
unknown option --show-column-numbers
usage: /usr/bin/python3 [option] ... [-c cmd | -m mod | file | -] [arg] ...
Try `python -h' for more information.

Compilation exited abnormally with code 2 at Fri Jul 31 21:21:40
```

**Comment by cpitclaudel** [comment]
Thanks, that's very useful (the part I wanted to see was the command invocation; that wasn't in your original comment, was it? Sorry if I missed something.

**Comment by cpitclaudel** [comment]
Did you customize flycheck-python-mypy-executable ? Does the problem happen when you start in emacs -Q?

**Comment by xuhdev** [comment]
Ah ha, I did set that! For some reason it was set to `python3`. Weird though it would work in a virtual environment. Anyway, this issue should be resolved now. Thanks for your help!

**Comment by cpitclaudel** [comment]
It's probably our fault, at least in part: for the other python checkers you can set the checker name to `python` or `python3` and it will do the right thing.

**Comment by xuhdev** [comment]
Hmm... What would you say? I'm happy to provide any help you need from me.

**Comment by cpitclaudel** [comment]
I think it's fine to leave it like this: python2 is deprecated and mypy is only useful for python3.
